TheEthicalHackerCup/Assets/AttackMinigames/DDoSAttack/Scripts/SpawnFinishLine.cs

```diff
@@ -0,0 +1,22 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class SpawnFinishLine : MonoBehaviour
+{
+    public GameObject finishLine;
+    public float spawnTime;
```

Comment:
Set this up programmatically in a Start() method so that it can be used as a difficulty control. Later on, GameManager would have some functionality to return a difficulty level that would be used to dynamically set the spawn time for the finish line

Reply:
I found out why the finish line doesn't spawn on 2nd attempts of the game. The if statement in the is comparing the timeSinceGameStart with the spawnTime which is something like 10. On subsequent attempts, the timeSinceGameStart is will be bigger than the spawnTime variable so it'll never get into the if statement.

Fix is reasonably simple as shown in the screenshot. Just add a start method that tracks the time that the minigame starts and then compare the absolute time to the minigameStartTime+timeUntilSpawn

I'd recommend using framecount instead of fixedTime and changing your var to framesUntilSpawn so that you'd be working with ints instead of floats. That can avoid potential never-spawn issues with float comparison. (Assume 30fps)

Using ints also avoids the double finish line spawn issue. I suspect that this is be because Time.fixedTime gets updated every other frame, thus allowing us to get into the if statement twice



![image](https://user-images.githubusercontent.com/72239525/214197896-fb744a77-11cb-4b10-a841-6c724f06be56.png)